src/textual/app.py

```diff
@@ -319,28 +324,33 @@ def bind(
             keys, action, description, show=show, key_display=key_display
         )
 
-    def run(self, loop: AbstractEventLoop | None = None) -> ReturnType | None:
+    def run(self) -> ReturnType | None:
         async def run_app() -> None:
             await self.process_messages()
 
-        if loop:
-            asyncio.set_event_loop(loop)
-        else:
-            try:
-                import uvloop
-            except ImportError:
-                pass
-            else:
-                asyncio.set_event_loop(uvloop.new_event_loop())
-
-        event_loop = asyncio.get_event_loop()
-        try:
+        if _ASYNCIO_GET_EVENT_LOOP_IS_DEPRECATED:
+            # N.B. This doesn't work with Python<3.10, as we end up with 2 event loops:
             asyncio.run(run_app())
-        finally:
-            event_loop.close()
+        else:
+            # However, this works with Python<3.10:
+            event_loop = asyncio.get_event_loop()
+            event_loop.run_until_complete(run_app())
 
         return self._return_value
 
+    @classmethod
+    def _init_uvloop(cls) -> None:
+        if hasattr(cls, "__uvloop_installed"):
```

Comment:
No sure this is necessary. I think that `uvloop.install` may be a no-op when called twice. Can you check?

Reply:
I checked before writing this (ugly :sweat_smile: ) code, yes, and it seems that calling it twice "installs" it twice.
> https://github.com/MagicStack/uvloop/blob/master/uvloop/__init__.py#L23-L25

I reckon that executing this `__asyncio.set_event_loop_policy(EventLoopPolicy())` statement twice doesn't harm, but I don't know how costly replacing the event loop policy multiple times is for Python in terms of performance?

Reply:
Fair enough. Still I think we should store that boolean somewhere else, where it can be typed and doesn't need `hasattr`.

How about a classvar?

Reply:
As discussed, I moved this logic to the module scope : [e324de7](https://github.com/Textualize/textual/pull/449/commits/e324de761306eed5252b9189aa8a132ac7378dac) :slightly_smiling_face: